src/db.rs

```diff
@@ -56,6 +57,47 @@ const KEEP_LOGS: usize = 16;
 /// Value is just a vector of bytes. Value sizes up to 4Gb are allowed.
 pub type Value = Vec<u8>;
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValuePtr(Arc<Value>);
```

Comment:
I would also maybe just consider a type alias like `type RcKey = Arc<Key>; type RcValue = Arc<Value>` to reduce the boilerplate implementation.

Reply:
Renaming done